docs/source-app/workflows/add_web_ui/panel/panel_frontend.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import sys
+
+import panel as pn
+
+import lightning_app as lapp
+from lightning_app.utilities.imports import requires
+
+logger = logging.getLogger("PanelFrontend")
+logger.setLevel(logging.DEBUG)
+
+handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter("%(asctime)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+class PanelFrontend(lapp.LightningWork):

```

Comment:
@tchaton. Here I need to figure out the right class to inherit from. I need to be able to share share and run on GPUs. As far as I know the `Frontend` that `StreamlitFrontend` inherits from does not support this. I can see that `ServeGradio` inherits from `LightningWork`.